diff --git a/CMakeLists.txt b/CMakeLists.txt
index 81d40ef..c7d1b13 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -125,6 +125,9 @@ if(BYOD_BUILD_CLAP)
     )
 endif()
 
+# this has to be placed before juce_generate_juce_header()...
+juce_add_modules(${AAP_JUCE_DIR}/aap-modules/aap_audio_processors)
+
 juce_generate_juce_header(BYOD)
 add_subdirectory(src)
 include_directories(src)
@@ -146,3 +149,43 @@ option(BYOD_BUILD_ADD_ON_MODULES "Build BYOD with add-on modules" OFF)
 if(BYOD_BUILD_ADD_ON_MODULES)
     add_subdirectory(modules/BYOD-add-ons)
 endif()
+
+# begin AAP specifics -->
+
+# They are needed in desktop too, for generate-aap-metadata.
+target_include_directories(BYOD PUBLIC
+        "${AAP_DIR}/include"
+        "${AAP_JUCE_DIR}/aap-modules"
+)
+
+target_compile_definitions(BYOD PUBLIC
+        JUCEAAP_HAVE_AUDIO_PLAYHEAD_NEW_POSITION_INFO=1
+        JUCE_PUSH_NOTIFICATIONS=1
+        JUCE_PUSH_NOTIFICATIONS_ACTIVITY="com/rmsl/juce/JuceActivity"
+        )
+
+message("AAP_DIR: ${AAP_DIR}")
+message("AAP_JUCE_DIR: ${AAP_JUCE_DIR}")
+
+if (ANDROID)
+    find_library(log "log")
+    find_library(android "android")
+    find_library(glesv2 "GLESv2")
+    find_library(egl "EGL")
+    #set(cpufeatures_lib "cpufeatures")
+    #set(oboe_lib "oboe")
+
+    target_include_directories(BYOD PRIVATE
+            "${ANDROID_NDK}/sources/android/cpufeatures"
+            #"${OBOE_DIR}/include"
+            )
+
+    add_compile_definitions(
+            [[JUCE_ANDROID=1]]
+            [[JUCE_DONT_AUTO_OPEN_MIDI_DEVICES_ON_MOBILE=1]]
+    )
+endif (ANDROID)
+
+target_link_libraries(BYOD PRIVATE aap_audio_processors)
+
+# <-- end AAP specifics
diff --git a/modules/CMakeLists.txt b/modules/CMakeLists.txt
index 88bf159..e929083 100644
--- a/modules/CMakeLists.txt
+++ b/modules/CMakeLists.txt
@@ -72,6 +72,8 @@ target_compile_definitions(juce_plugin_modules
         JucePlugin_VersionString="${CMAKE_PROJECT_VERSION}"
         JucePlugin_Name="${CMAKE_PROJECT_NAME}"
         JUCE_MODAL_LOOPS_PERMITTED=1
+        JUCE_PUSH_NOTIFICATIONS=1
+        JUCE_PUSH_NOTIFICATIONS_ACTIVITY="com/rmsl/juce/JuceActivity"
     INTERFACE
         $<TARGET_PROPERTY:juce_plugin_modules,COMPILE_DEFINITIONS>
 )
Submodule modules/chowdsp_utils contains modified content
diff --git a/modules/chowdsp_utils/modules/gui/chowdsp_gui/InfoUtils/chowdsp_SystemInfo.h b/modules/chowdsp_utils/modules/gui/chowdsp_gui/InfoUtils/chowdsp_SystemInfo.h
index 106c005..ddedf3b 100644
--- a/modules/chowdsp_utils/modules/gui/chowdsp_gui/InfoUtils/chowdsp_SystemInfo.h
+++ b/modules/chowdsp_utils/modules/gui/chowdsp_gui/InfoUtils/chowdsp_SystemInfo.h
@@ -18,6 +18,8 @@ namespace SystemInfo
         return "Mac";
 #elif JUCE_IOS
         return "IOS";
+#elif JUCE_ANDROID
+    return "Android";
 #elif JUCE_LINUX
 #if JUCE_64BIT
         return "Linux64";
diff --git a/modules/chowdsp_utils/modules/gui/chowdsp_gui/Presets/chowdsp_PresetsComp.cpp b/modules/chowdsp_utils/modules/gui/chowdsp_gui/Presets/chowdsp_PresetsComp.cpp
index 707e023..ac0c516 100644
--- a/modules/chowdsp_utils/modules/gui/chowdsp_gui/Presets/chowdsp_PresetsComp.cpp
+++ b/modules/chowdsp_utils/modules/gui/chowdsp_gui/Presets/chowdsp_PresetsComp.cpp
@@ -256,11 +256,11 @@ int PresetsComp::addSavePresetOptions (int optionID)
                                                   auto presetFile = curPreset->getPresetFile();
                                                   if (! (presetFile.existsAsFile() && presetFile.hasFileExtension (presetExt)))
                                                   {
-                                                      juce::NativeMessageBox::showMessageBox (juce::MessageBoxIconType::WarningIcon, "Preset Deletion", "Unable to find preset file!");
+                                                      //juce::NativeMessageBox::showMessageBox (juce::MessageBoxIconType::WarningIcon, "Preset Deletion", "Unable to find preset file!");
                                                       return;
                                                   }
 
-                                                  if (juce::NativeMessageBox::showOkCancelBox (juce::MessageBoxIconType::QuestionIcon, "Preset Deletion", "Are you sure you want to delete this preset? This operation cannot be undone."))
+                                                  if (true)//juce::NativeMessageBox::showOkCancelBox (juce::MessageBoxIconType::QuestionIcon, "Preset Deletion", "Are you sure you want to delete this preset? This operation cannot be undone."))
                                                   {
                                                       presetFile.deleteFile();
                                                       manager.loadDefaultPreset();
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f44c119..c9ed1d4 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -206,5 +206,5 @@ file(GLOB_RECURSE juce_module_sources
 set_source_files_properties(${juce_module_sources}
     TARGET_DIRECTORY BYOD
     PROPERTIES SKIP_PRECOMPILE_HEADERS TRUE SKIP_UNITY_BUILD_INCLUSION TRUE)
-set_target_properties(BYOD PROPERTIES UNITY_BUILD ON UNITY_BUILD_BATCH_SIZE 8)
-target_precompile_headers(BYOD PRIVATE pch.h)
+#set_target_properties(BYOD PROPERTIES UNITY_BUILD ON UNITY_BUILD_BATCH_SIZE 8)
+#target_precompile_headers(BYOD PRIVATE pch.h)
diff --git a/src/gui/utils/ErrorMessageView.cpp b/src/gui/utils/ErrorMessageView.cpp
index 19cdc2f..5df0d2c 100644
--- a/src/gui/utils/ErrorMessageView.cpp
+++ b/src/gui/utils/ErrorMessageView.cpp
@@ -81,8 +81,10 @@ bool ErrorMessageView::showYesNoBox (const juce::String& title, const juce::Stri
         errorMessageView.setParametersYesNo (title, message);
         errorMessageView.setVisible (true);
 
+#if !ANDROID
         while (errorMessageView.result < 0)
             MessageManager::getInstance()->runDispatchLoopUntil (50);
+#endif
 
         return errorMessageView.result > 0;
     }
diff --git a/src/headless/CMakeLists.txt b/src/headless/CMakeLists.txt
index e580113..11820a6 100644
--- a/src/headless/CMakeLists.txt
+++ b/src/headless/CMakeLists.txt
@@ -16,19 +16,19 @@ target_sources(BYOD_headless PRIVATE
     ScreenshotGenerator.cpp
     GuitarMLFilterDesigner.cpp
 
-    tests/AmpIRsSaveLoadTest.cpp
+    #tests/AmpIRsSaveLoadTest.cpp
     tests/BadModulationTest.cpp
-    tests/ForwardingParamStabilityTest.cpp
+    #tests/ForwardingParamStabilityTest.cpp
     tests/NaNResetTest.cpp
     tests/ParameterSmoothTest.cpp
     tests/PreBufferTest.cpp
-    tests/PresetsTest.cpp
+    #tests/PresetsTest.cpp
     tests/PresetSearchTest.cpp
     tests/ProcessorStoreInfoTest.cpp
     tests/RAMUsageTest.cpp
     tests/SilenceTest.cpp
-    tests/StereoTest.cpp
-    tests/UndoRedoTest.cpp
+    #tests/StereoTest.cpp
+    #tests/UndoRedoTest.cpp
     tests/UnitTests.cpp
     tests/WaveshaperTest.cpp
 )
diff --git a/src/processors/chain/ProcessorChainActions.cpp b/src/processors/chain/ProcessorChainActions.cpp
index f864796..a35aaa9 100644
--- a/src/processors/chain/ProcessorChainActions.cpp
+++ b/src/processors/chain/ProcessorChainActions.cpp
@@ -127,7 +127,9 @@ bool waitForPointerCheck (const PointerType& pointer, int waitCycles = 6)
         if (count >= waitCycles)
             return false;
 
+#if !ANDROID
         juce::MessageManager::getInstance()->runDispatchLoopUntil (50);
+#endif
         count++;
     }
 
diff --git a/src/processors/drive/junior_b/NeuralTriodeModel.h b/src/processors/drive/junior_b/NeuralTriodeModel.h
index f4cca50..df5cbfb 100644
--- a/src/processors/drive/junior_b/NeuralTriodeModel.h
+++ b/src/processors/drive/junior_b/NeuralTriodeModel.h
@@ -94,7 +94,7 @@ template <typename T, typename ModelType>
 class NeuralTriodeModel
 {
 public:
-#if JUCE_MAC || JUCE_WINDOWS || JUCE_LINUX || JUCE_IOS
+#if JUCE_MAC || JUCE_WINDOWS || JUCE_LINUX || JUCE_IOS || JUCE_ANDROID
     explicit NeuralTriodeModel (const char* modelData, int modelDataSize)
     {
 #if JUCE_DEBUG
